rust/signed_doc/src/lib.rs

```diff
@@ -1 +1,192 @@
 //! Catalyst documents signing crate
+#![allow(dead_code)]
+use std::{
+    convert::TryFrom,
+    fmt::{Display, Formatter},
+    sync::Arc,
+};
+
+use coset::{CborSerializable, TaggedCborSerializable};
+
+mod metadata;
+
+pub use metadata::{DocumentRef, Metadata, UuidV7};
+
+/// Catalyst Signed Document Content Encoding Key.
+const CONTENT_ENCODING_KEY: &str = "content encoding";
```

Comment:
We should mirror HTTP here, not define a new spec (unless there is a spec for this I am missing).
See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding
```suggestion
const CONTENT_ENCODING_KEY: &str = "Content-Encoding";
```
We should also be case insensitive.  So `Content-Encoding` is the same as `content-encoding`.
If `Content-Encoding` is absent, just like HTTP, then there is no encoding of the content, it's raw.